### DISCUSSION TITLE
Sujet / Thema 14 –– Lückenlose, nicht überlappende Bodenbedeckung / Couverture du territoire sans interruption et sans superposition

### DISCUSSION THREAD

**Discussion by ratnanil** [GeoUnconference/discussions#32] [Q&A]
Was ich im SwissTLM3D Datensatz immer wieder vermisse ist ein lückenloser, nicht überlappender Vektor Datensatz der Bodenbedeckung (analog [Vector25 Primärflächen](https://map.geo.admin.ch/?topic=swisstopo&lang=de&bgLayer=voidLayer&X=198208.10&Y=645051.44&zoom=2&layers=ch.swisstopo.vec25-primaerflaechen&catalogNodes=1436)).  Im Datensatz _SwissTLM3D-Bodenbedeckung_ fehlen beispielsweise Siedlungs- und Landwirtschaftsflächen. Landwirtschaftsflächen kann ich aus den Kantonalen Daten (z.B. [Nutzungsflächen](https://geodienste.ch/services/lwb_nutzungsflaechen))  entnehmen, doch das Siedlungsgebiet bietet mir etwas Schwierigkeiten. Ich sehe folgende Möglichkeiten

- Berechnung via einer Aggregation den Gebäudegrundrissen (rechnerisch sehr aufwendig)
- Verwendung von [Vector25 Primärflächen](https://map.geo.admin.ch/?topic=swisstopo&lang=de&bgLayer=voidLayer&X=198208.10&Y=645051.44&zoom=2&layers=ch.swisstopo.vec25-primaerflaechen&catalogNodes=1436) (diese sind aber hoffnungslos veraltet)
- Verwendung von [SwissTLM Region](https://map.geo.admin.ch/?topic=swisstopo&lang=de&bgLayer=ch.swisstopo.swissimage&zoom=9&catalogNodes=1436&layers=ch.swisstopo.vec200-landcover&E=2683375.79&N=1238247.99&layers_opacity=0.65)
- Verwendung der kantonalen, [Kommunalen Nutzungsplanung](https://geodienste.ch/services/npl_nutzungsplanung) (nicht Schweizweit verfügbar)

Gibt es noch andere Ideen? Oder hat bereits jemand eine lückenlose, nicht überlappende Bodenbedeckung für die Schweiz entwickelt?

**Comment by RafBov** [comment]
Regarding the "Siedlungsgebiet". swisstopo derived a layer from the builidings footprints for the Bundesamt für Zivilluftfahrt. Bebaute Gebiete nach Schweizer Luftfahrtrecht (Bundesamt für Zivilluftfahrt (BAZL)) [https://s.geo.admin.ch/96b3228abd](https://s.geo.admin.ch/96b3228abd) . This is of course a dataset that shows the built-up area based on existing buildings. It has nothing to do with potential building zones.  Different rules (minimum areas, etc..) have been put in place to generate the layer. swisstopo has thought about making this layer derived from the TLM directly available every year. We have not done so for the moment, because we are not sure of the demand of users.  Would such a layer help you or others ? 

\* fixed URL

**Reply by rastrau** [reply]
I would find this dataset interesting. Currently, we sometimes use the `Ort` category in swissTLM3D as a proxy for settlement areas. The BAZL-oriented dataset looks more organic and may work well for some applications.

**Comment by ratnanil** [comment]
I would be interested as well! As @rastrau commented, the dataset looks very organic and useable. 

BTW: I'm not done creating a seamless vector data set for Switzerland. My initial approach creating this via raster data is not as trivial as I thought (check [this Stackoverflow question ](https://gis.stackexchange.com/q/424008/40929) and [this twitter thread](https://twitter.com/NRatnaweera/status/1493890501921095685)). I thought of copying the [Corine Landcover](https://www.eea.europa.eu/data-and-maps/data/copernicus-land-monitoring-service-corine) workflow, but as a user commented: 

> Being quite familiar with the production process of the corine landcover dataset, it is unfortunately not as easy as you think, and there is no one-liner solution ;)

**Reply by rastrau** [reply]
Looking at https://github.com/GeoUnconference/discussions/discussions/32#discussion-3832682, (I'm sure I'm missing something) why don't you follow a pure-vector approach combining parts of swissTLM<sup>3D</sup>, Nutzungsflächen (not familiar with these, are these vector?) and settlement areas as vectors (either from swissTLM<sup>3D</sup>, from CORINE LC or from @RafBov's dataset – all of which are vector)? To me, this seems less complicated than ensuring planar enforcement through rasterization-and-vectorization.

**Reply by ratnanil** [reply]
The problem with integrating these data sources with the vector approach is, that the resulting layers contains gaps and overlaps. Fixing these in the vector world is cumbersome and computationally expensive. The raster world has out of the box solutions for these situations:  overlaps can simply be `gdal`-burned over each other, gaps are easily determined through `NA` values, slivers are removed with   `gdal_sieve.py` etc. 

**However** the price for switch between the two worlds seems to be too high, so I'm now following the vector approach as you suggest.

**Reply by rastrau** [reply]
I see. I'm sure I imagine it much too simple. Currently, I'm conceptualizing it as taking a planar-enforced subset of swissTLM<sup>3D</sup> and adding settled areas as well as agricultural areas, strictly suppressing what swissTLM<sup>3D</sup> contains in the affected locations. This would roughly amount to, in Esri terms, Erase and Append, or Union and postprocessing. Potential slivers could be automatically merged with neighbouring polygons (although with rules that are so simple that they approach arbitrary merging).
